src/BlazorWebView/src/SharedSource/ExternalLinkNavigationEventArgs.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+namespace Microsoft.AspNetCore.Components.WebView
+{
+	/// <summary>
+	/// Used to provide information about a link (<![CDATA[<a>]]>) clicked within a Blazor WebView.
+	/// 
+	/// `_blank` target links will always open in the default browser
+	/// thus the OnExternalNavigationStarting won't be called.
+	/// </summary>
+	public class ExternalLinkNavigationEventArgs : EventArgs
+	{
+		public ExternalLinkNavigationEventArgs(Uri uri)
+		{
+			Uri = uri;
+		}
+
+		/// <summary>
+		/// External <see cref="Uri">URI</see> to be navigated to.
+		/// </summary>
+		public Uri Uri { get; set; }
```

Comment:
Should this have a setter? I think not?

Reply:
I don't see any harm in having it? Wanted to support the hypothetical use case of redirecting the navigation request based on the existing Uri.

Reply:
Doesn't seem necessary because that isn't the point of this. If someone wants a link to go somewhere else, they should just do that to begin with. Also, the current code doesn't even always work that way with a modified event args URL because in at least some cases it uses the original URL and not the one from the event args that may have been modified:

https://github.com/dotnet/maui/pull/4645/files#diff-75c4562a2bc68615b8d28c071b33a112272d1e08e4ba087c461736787ff8f534R199